docs/Explore Algorithms/OpenAI/Quickstart - Custom Embeddings and Approximate KNN.ipynb

```diff
@@ -0,0 +1,550 @@
+    "    file_path\n",
+    ")\n",
+    "df = df.withColumn(\n",
+    "    \"combined\",\n",
+    "    F.format_string(\n",
+    "        \"Title: %s; Content: %s\", F.trim(df.Summary), F.trim(df.Text)\n",
+    "    ),\n",
+    ")\n",
+    "\n",
+    "# Size of DF\n",
+    "number_of_input_rows = 999\n",
+    "# Shuffle the DataFrame with a fixed seed\n",
+    "seed = 42\n",
+    "\n",
+    "# Check if the row count is less than 10\n",
+    "if number_of_input_rows <= 0 or number_of_input_rows >= 1000000:\n",
+    "    raise ValueError(f\"Limit is {number_of_input_rows}, which should be less than 1M.\")\n",
+    "\n",
+    "if number_of_input_rows > 1000:\n",
+    "\n",
+    "    # Cross-join the DataFrame with itself to create n x n pairs for string concatenation (synthetic data)\n",
+    "    cross_joined_df = df.crossJoin(\n",
+    "        df.withColumnRenamed(\"combined\", \"combined_\")\n",
+    "    )\n",
+    "\n",
+    "    # Create a new column 'result_vector' by concatenating the two source vectors\n",
+    "    tmp_df = cross_joined_df.withColumn(\n",
+    "        \"result_vector\",\n",
+    "        F.concat(F.col(\"combined\"), F.lit(\". \\n\"), F.col(\"combined_\")),\n",
+    "    )\n",
+    "\n",
+    "    # Select only the necessary columns and show the result\n",
+    "    tmp_df = tmp_df.select(\"result_vector\")\n",
+    "    df = tmp_df.withColumnRenamed(\"result_vector\", \"combined\").withColumn(\n",
+    "        \"id\", monotonically_increasing_id()\n",
+    "    )\n",
+    "\n",
+    "df = df.limit(number_of_input_rows).orderBy(rand(seed)).repartition(10).cache()\n",
+    "\n",
+    "print(f\"Loaded: {number_of_input_rows} rows\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "0c69ee56-172f-413b-a335-d15482fda55e",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Step 3: Generate Embeddings\n",
+    "\n",
+    "We will first generate embeddings using NVIDIA TensorRT optimized SentenceTransformer. In the demo you can use two fifferent HF models: intfloat/e5-large-v2 or sentence-transformers/all-MiniLM-L6-v2\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 0,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "8d7bd9db-79a1-4d46-a849-ac49c3de7b49",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# dataTransformer = HuggingFaceSentenceEmbedder(modelName=\"intfloat/e5-large-v2\", inputCol=\"combined\", outputCol=\"embeddings\", runtime=\"tensorrt\")\n",
+    "dataTransformer = HuggingFaceSentenceEmbedder(modelName=\"sentence-transformers/all-MiniLM-L6-v2\", inputCol=\"combined\", outputCol=\"embeddings\", runtime=\"tensorrt\")\n",
```

Comment:
nit: make model name a param, and then pass model name, give people a few to try